activerecord/test/cases/statement_cache_test.rb

```diff
@@ -105,5 +105,18 @@ def test_unprepared_statements_dont_share_a_cache_with_prepared_statements
 
       refute_equal book, other_book
     end
+
+    def test_tablename
+      book = Book.create(name: "my book")
+      assert_equal book, Book.find_by(name: "my book")
```

Comment:
Do we expect this finder to fail? Would limit assertions to just the expectations that would fail if this PR were reverted.